File: library/src/main/java/com/trinity/decoder/Mp3Decoder.java
package com.trinity.decoder;

public interface Mp3Decoder {

	public void init(float packetBufferTimePercent, int vocalSampleRate);

	public void setAccompanyVolume(float volume, float accompanyMax);
	
	public void destory();

	public int readSamples(short[] samples, int[] slientSizeArr, int[] extraSampleType);

	public void startAccompany(String path);
	public void pauseAccompany();
	public void resumeAccompany();
	public void stopAccompany();
}

File: library/src/main/java/com/trinity/decoder/Mp3Decoder.kt
package com.trinity.decoder

interface Mp3Decoder {

    fun init(packetBufferTimePercent: Float, vocalSampleRate: Int)
    fun setAccompanyVolume(volume: Float, accompanyMax: Float)
    fun destroy()
    fun readSamples(samples: ShortArray, slientSizeArr: IntArray, extraSampleType: IntArray): Int
    fun startAccompany(path: String)
    fun pauseAccompany()
    fun resumeAccompany()
    fun stopAccompany()
}
